# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/vm/Xdr.cpp
# Commit: d4359a64d30f
# Full Hash: d4359a64d30fe311f0d3aff7b0e50e57874a0cd2
# Author: caroline <cullen.caroline@gmail.com>
# Date: 2019-12-20 21:54:48
# Description:
#   Bug 1604747 - Error if attempt at encoding and instantiated module. r=iain
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D57900
# ==============================================================================

diff -r 0480be98e2a9 -r d4359a64d30f js/src/vm/Xdr.cpp
--- a/js/src/vm/Xdr.cpp	Fri Dec 20 16:27:08 2019 +0000
+++ b/js/src/vm/Xdr.cpp	Fri Dec 20 16:38:35 2019 +0000
@@ -228,6 +228,8 @@
 #endif
   if (mode == XDR_DECODE) {
     modp.set(nullptr);
+  } else {
+    MOZ_ASSERT(modp->status() < MODULE_STATUS_INSTANTIATING);
   }
 
   MOZ_TRY(XDRModuleObject(this, modp));
